### PR TITLE
Circleci project setup

### LINKED ISSUE #4: Add Circle CI
This issue can be closed when there is a Circle CI build that runs pytest and flake8 with every PR

### PR DESCRIPTION
**Pull request by samgdotson**
This PR adds:
- CircleCI to the project (for automatic execution of tests)
- ``requirements.txt`` file 

Closes #4

### CODE REVIEW

**Comment by robfairh** [comment]
Why are you making the PR from arfc:circleci-project-setup ?

**Comment by samgdotson** [comment]
The file was created from the circle-ci website. The service created its own branch through github and is making the PR.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          keys:
+          - v1-dependencies-{{ checksum "requirements.txt" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+      - run:
+          name: requirements
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
```
Comment (robfairh) [inline .circleci/config.yml:24]: why using an extra file when you could define this in one line?
pip install numpy matplotlib pandas pytest

Comment (samgdotson) [reply .circleci/config.yml:24]: It's easier to update this file to make sure the requirements are met automatically. I want to minimize the amount I have to edit the ``config.yml`` file

Comment (robfairh) [reply .circleci/config.yml:24]: Sounds good.

**Pull request merged**